tests/src/test/scala/pureconfig/EnumerationsSuite.scala

```diff
@@ -72,6 +72,9 @@ class EnumerationsSuite extends BaseSuite {
   }
 
   it should "not allow deriving readers, writers and full converters for enumerations encoded as sealed traits whose subclasses are not all case objects" in {
```

Comment:
Can we have a test checking that we disallow derivation of enum readers for case classes at compile time, like this one? I'm ok with ignoring the zero-arg case class edge case, but I'd like to ensure we don't generate type classes for case classes with arguments.

Reply:
Added a copy of this example to pureconfig-magnolia's test suite in b02e5dc.